gcc/rust/lex/rust-lex.cc

```diff
@@ -186,6 +202,50 @@ Lexer::skip_input ()
   skip_input (0);
 }
 
+void
+Lexer::skip_token (int n)
+{
+  // dump tokens if dump-lex option is enabled
+  if (dump_lex_enabled && !dump_lex_out.fail ())
```

Comment:
`dump_lex_out.fail()` should not be checked here imo. If it fails to open, it's a compiler error and we should trigger an assertion in the constructor

Reply:
fixed